Sources/URLNavigator/NavigatorType.swift

```diff
@@ -86,19 +86,22 @@ extension NavigatorType {
   public func handler(for url: URLConvertible) -> URLOpenHandler? {
     return self.handler(for: url, context: nil)
   }
-
+  
   @discardableResult
-  public func pushURL(_ url: URLConvertible, context: Any? = nil, from: UINavigationControllerType? = nil, animated: Bool = true) -> UIViewController? {
+  public func pushURL(_ url: URLConvertible, context: Any? = nil, from: UINavigationControllerType? = nil, animated: Bool = true, completion: (() -> Void)? = nil) -> UIViewController? {
     guard let viewController = self.viewController(for: url, context: context) else { return nil }
-    return self.pushViewController(viewController, from: from, animated: animated)
+    return self.pushViewController(viewController, from: from, animated: animated, completion: completion)
   }
-
+  
   @discardableResult
-  public func pushViewController(_ viewController: UIViewController, from: UINavigationControllerType?, animated: Bool) -> UIViewController? {
+  public func pushViewController(_ viewController: UIViewController, from: UINavigationControllerType?, animated: Bool, completion: (() -> Void)?) -> UIViewController? {
     guard (viewController is UINavigationController) == false else { return nil }
     guard let navigationController = from ?? UIViewController.topMost?.navigationController else { return nil }
     guard self.delegate?.shouldPush(viewController: viewController, from: navigationController) != false else { return nil }
+    CATransaction.begin()
+    CATransaction.setCompletionBlock(completion)
```

Comment:
Is this trick safe?

Reply:
it's safe, i reference from SwifterSwift library.